src/melee/ft/chara/ftSeak/ftSeak_SpecialS.c

```diff
@@ -0,0 +1,1104 @@
+
+    {
+        s32 stateVar3 = fp->x2348_stateVar3_s32;
+
+        if (stateVar3 > 0) {
+            fp->x2348_stateVar3_s32 = stateVar3 - 1;
+        } else {
+            const enum_t flags = (1 << 3) | (1 << 6) | (1 << 8) | (1 << 9) |
+                                 (1 << 10) | (1 << 11) | (1 << 12) | (1 << 18);
+
+            if ((fp->facing_dir == +1 && fp->sa.seak.lstick_delta.x > +0.3F) ||
+                (fp->facing_dir == -1 && fp->sa.seak.lstick_delta.x < -0.3F))
+            {
+                func_80088148(fp, flags, 127, 64);
+                fp->x2348_stateVar3_s32 = 6;
+            } else if (fp->sa.seak.lstick_delta.y > 0.5F) {
+                func_80088148(fp, flags, 127, 64);
+                fp->x2348_stateVar3_s32 = 12;
+            }
+        }
+    }
+
+    {
+        s32 stateVar4 = fp->x234C_stateVar4_s32;
+        if (stateVar4 > 0) {
+            fp->x234C_stateVar4_s32 = stateVar4 - 1;
+        } else {
+            const enum_t flags = (1 << 0) | (1 << 1) | (1 << 3) | (1 << 6) |
+                                 (1 << 8) | (1 << 9) | (1 << 10) | (1 << 11) |
+                                 (1 << 12) | (1 << 18);
+
+            if ((fp->facing_dir == +1 && fp->sa.seak.lstick_delta.x < -0.3F &&
+                 fp->input.x620_lstick_x < 0) ||
+                (fp->facing_dir == -1 && fp->sa.seak.lstick_delta.x > +0.3F &&
+                 fp->input.x620_lstick_x > 0))
+            {
+                func_80088148(fp, flags, 127, 64);
+                fp->x234C_stateVar4_s32 = 6;
+            }
+        }
+    }
+
+    {
+        HSD_GObj* item_gobj = fp->sa.seak.x2234;
+
+        if (item_gobj == NULL)
+            return;
+
+        {
+            f32 left_stick_x = fp->input.x620_lstick_x;
+            Item* item_data = item_gobj->user_data;
+            Article* article = item_data->xC4_article_data;
+            itChainSegment* chainSegment = article->x4_specialAttributes;
+
+            if (left_stick_x < 0)
+                left_stick_x = -left_stick_x;
+
+            if (left_stick_x < chainSegment->x48) {
+                f32 left_stick_y = fp->input.x624_lstick_y;
+
+                if (fp->input.x624_lstick_y < 0)
+                    left_stick_y = -left_stick_y;
+
+                if (left_stick_y < chainSegment->x48) {
+                    f32 mul = 0.5F;
+                    fp->sa.seak.lstick_delta.x *= mul;
+                    fp->sa.seak.lstick_delta.y *= mul;
+                }
+            }
+        }
+    }
+}
+
+void ftSeak_8011097C(HSD_GObj* fighter_gobj)
+{
+    return;
+}
+
+void ftSeak_UpdateHitboxes(HSD_GObj* fighter_gobj, Vec3* new_pos, s32 hitbox_id)
+{
+    if (fighter_gobj == NULL)
+        return;
+
+    {
+        Fighter* fp = GET_FIGHTER(fighter_gobj);
+
+        if (!fp->x2200_ftcmd_var0)
+            return;
+
+        {
+            u8* someptr = (u8*) fp + (12 * hitbox_id);
+            __memcpy(&someptr[0x2238], new_pos, 12);
```

Comment:
Going back and forth on whether we should fix these before merging. Should clearly be copying new_pos to a Vec3 in fp's special attrs indexed by hitbox_id, same for the memcpys below. Probably not a big deal if we merge and clean it up later.

Reply:
Does `Fighter` already have all the necessary fields to do this? If so we should fix it now.

Reply:
I didn't like that either. I tried a few different syntax permutations to perform that copy, but ultimately that was the only one I could find that matched.

Reply:
```suggestion
           fp->sa.seak.x2238[hitbox_id] = *new_pos;
```